src/workerd/io/actor-sqlite.c++

```diff
@@ -169,11 +200,67 @@ void ActorSqlite::onWrite() {
       // rather than after the callback.
       { auto drop = kj::mv(txn); }
 
-      return commitCallback();
+      return commitImpl(kj::mv(precommitAlarmPromise));
     })));
   }
 }
 
+kj::Promise<void> ActorSqlite::requestScheduledAlarm(kj::Maybe<kj::Date> requestedTime) {
+  co_await hooks.scheduleRun(requestedTime);
```

Comment:
Hmm I don't remember... if `scheduleRun()` throws synchronously, will `requestScheduleAlarm()` throw synchronously, or does the co_await catch the exception and turn it into a rejected promise?

We need the exception to flow through synchronously in the workerd case (so that the transaction is never committed). It might be best not to use coroutines here, to be safe.

Reply:
Wow, good catch... the coroutine here was indeed turning synchronously thrown exceptions into rejected promises.